R/data.table.R

```diff
@@ -968,7 +968,39 @@ replace_dot_alias = function(e) {
         }
         # fix for long standing FR/bug, #495 and #484
         allcols = c(names_x, xdotprefix, names_i, idotprefix)
-        if ( length(non_sdvars <- setdiff(intersect(av, allcols), c(bynames, ansvars))) ) {
+        non_sdvars <- setdiff(intersect(av, allcols), c(bynames, ansvars))
+        
+        # added 'mget' - fix for #994
+        if (any(c("get", "mget") %chin% av)){
+          # get('varname') is too difficult to detect which columns are used in general
+          # eval(macro) column names are detected via the  if jsub[[1]]==eval switch earlier above.
+          b_warn = FALSE 
+          if (is.call(jsub) && length(jsub[[1L]]) == 1L && jsub[[1L]] == ":="){
+            if (is.symbol(jsub[[2L]])) {
+              jsub_lhs_symbol = as.character(jsub[[2L]])
+              if (jsub_lhs_symbol %chin% non_sdvars) {
+                sdvars = setdiff(sdvars, jsub_lhs_symbol)
+              }
+            } else {
+              # fix for #4089 - if all names are already accounted for, we don't need to re-extract the names for the (m)get call
+              b_warn = is.null(names_i) && all(names_x %chin% c(ansvars, allbyvars))
+            }
+          }
+          if (!b_warn) {
+            if (verbose) 
+              cat("'(m)get' found in j. ansvars being set to all columns. Use .SDcols or a single j=eval(macro) instead. Both will detect the columns used which is important for efficiency.\nOld:", paste(ansvars,collapse=","),"\n")
```

Comment:
Please convert this to use `cat(gettextf(..., domain="R-data.table"))` so it's caught for message translation later

Reply:
You might want to use `brackify(ansvars)` as well